library/src/main/kotlin/build/buf/connect/impl/ProtocolClient.kt

```diff
@@ -52,8 +53,9 @@ class ProtocolClient(
         val serializationStrategy = config.serializationStrategy
         val requestCodec = serializationStrategy.codec(methodSpec.requestClass)
         try {
+            val httpURL = config.host.toHttpUrl().newBuilder().addPathSegment(methodSpec.path).build()
```

Comment:
The `config.host` value never changes, so why call `toHttpUrl` every time? Why not call it only once and store it so that you don't have the parsing overhead every time an endpoint is called?

Reply:
Yeah, that was probably not the right solution as it's an okhttp object